### PR TITLE
check if column is reserved and delete field from layout editor

### LINKED ISSUE #6119: [BUG] Reserved columns can be deleted from Layout Based Forms
### Describe the bug

isColumnReserved() is not stopping the removal of critical form fields such as fname, mname, lname, etc.

### To Reproduce
1. Go to Admin -> Forms -> Layouts
2. Go to Demographics
3. Select a reserved field (`lname` as an example)
4. Click Delete on the top under the "With:" menu
5. Confirm the alert box

### Expected behavior

Hard stop, preventing the user from deleting a critical form field

### Client configuration

- Browser: FF
- OpenEMR version: 7.0.1-dev
- Operating system: Development Easy Docker

### PR DESCRIPTION
**Pull request by stephenwaite**
Fixes #6119 and #6128

### CODE REVIEW

File: interface/super/edit_layout.php [hunk 1]
```diff
@@ -573,16 +573,22 @@ function encodeModifier($jsonArray)
...
     $comma = "";
+    $cntr = 0;
     foreach (explode(" ", $_POST['selectedfields']) as $onefield) {
-        $sqlstmt .= $comma . "'" . add_escape_custom($onefield) . "'";
-        $comma = ", ";
+        if (!isColumnReserved(tableNameFromLayout($_POST['layout_id']), $onefield)) {
+            $sqlstmt .= $comma . "'" . add_escape_custom($onefield) . "'";
+            $comma = ", ";
+            $cntr++;
+        }
```
Comment (sjpadgett) [inline interface/super/edit_layout.php:582]: imo the wrong place to do this. should be in the validation of rename or add function.
It's already there in code it's just there are some sql errors that when fix will fix issue.

I'll show where once I fix errors. cool

Comment (sjpadgett) [reply interface/super/edit_layout.php:582]: Sorry @stephenwaite you're correct I forgot the layouts table delete.
Still though there is another bug which is probably some of the recent layout editor issue.
L-287 all show table queries.

Comment (stephenwaite) [reply interface/super/edit_layout.php:582]: ok, nice, saw that, this adodb execute seems familiar, like something happened with that library

Comment (stephenwaite) [reply interface/super/edit_layout.php:582]: I think the bug is with non-alpha numeric named fields?

**Review by sjpadgett — APPROVED** [review]
This works for me with the `$tmp = sqlQuery("SHOW COLUMNS FROM `" . escape_table_name($tablename) . "` LIKE '$field_id'");` fix in the three places in script it is being used.

Thanks Stephen

**Review by bradymiller — APPROVED** [review]

**Comment by stephenwaite** [comment]
@bradymiller for #6128 , is the latest commit okay or should we investigate the escaping fail with mariadb?

File: interface/super/edit_layout.php [hunk 2]
```diff
@@ -284,7 +284,7 @@ function addOrDeleteColumn($layout_id, $field_id, $add = true)
         return;
     }
     // Check if the column currently exists.
-    $tmp = sqlQuery("SHOW COLUMNS FROM `" . escape_table_name($tablename) . "` LIKE ?", array($field_id));
+    $tmp = sqlQuery("SHOW COLUMNS FROM `" . escape_table_name($tablename) . "` LIKE '$field_id'");
```
Comment (bradymiller) [inline interface/super/edit_layout.php:287]: need to escape this via:
```
$tmp = sqlQuery("SHOW COLUMNS FROM `" . escape_table_name($tablename) . "` LIKE '" . add_escape_custom($field_id) . "'");
```

**Comment by bradymiller** [comment]
@stephenwaite , ok to work around it, but just need to escape the variables

File: interface/super/edit_layout.php (hunk 2 above)
Comment (bradymiller) [reply interface/super/edit_layout.php:287]: also need to do same with below lines

**Comment by bradymiller** [comment]
dance-bear

**Pull request merged**